core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorHandler.java

```diff
@@ -42,18 +40,24 @@
 
     private final ExceptionCatalog exceptionCatalog;
     private final ErrorMapper errorMapper;
```

Comment:
Okay, that's clearly better than keeping everything in one class.

However, reading the class now, I feel having a method like `handleWhitelistedException()` in this class, but having that method call `createExceptionParameters()` and `createErrorMessageDto()` in another class feels like a somewhat arbitrary separation of classes.

Let's continue moving out stuff of this class:
* `handleWhitelistedException()`, `handleClientError()` (incl `private determineErrorClass()`) and `handleUndefinedError()` move to `ErrorMapper`
* `handleConstraintViolationError()` moves to `ConstraintViolationMapper`
* `handleRuleViolationError()` moves to `RuleViolationMapper`